### PR TITLE
Update code_style.md

### PR DESCRIPTION
**Pull request by czeidler**
- Remove invalid single quotes item
- Add TypeScript section
- Add item regarding React function components and hooks

Signed-off-by: Clemens Zeidler <clemens.zeidler@gmail.com>

### CODE REVIEW

File: code_style.md [hunk 1]
```diff
@@ -164,6 +158,14 @@ ECMAScript
 - Apart from that, newer ES features should be used whenever the author deems them to be appropriate.
 - Flow annotations are welcome and encouraged.
```
Comment (swapnilraj) [inline code_style.md:159]: This should be probably be edited to encourage Typescript or deleted.

Comment (czeidler) [reply code_style.md:159]: Thanks I added a reference to TypeScript

File: code_style.md [hunk 2]
```diff
@@ -162,7 +156,15 @@ ECMAScript
   - Be careful mixing arrow functions and regular functions, eg. if one function in a promise chain is an
     arrow function, they probably all should be.
 - Apart from that, newer ES features should be used whenever the author deems them to be appropriate.
-- Flow annotations are welcome and encouraged.
+- While flow annotations are welcome and encouraged consider to migrate the JS file to TypeScript (see next section).
+
+TypeScript
+----------
+- TypeScript is preferred over the use of JavaScript
+- Its desirable to convert existing JavaScript files to TypeScript. TypeScript conversions should be done in small
```
Comment (jryans) [inline code_style.md:164]:
```suggestion
- It's desirable to convert existing JavaScript files to TypeScript. TypeScript conversions should be done in small
```

File: code_style.md [hunk 3]
```diff
@@ -162,7 +156,15 @@ ECMAScript
   - Be careful mixing arrow functions and regular functions, eg. if one function in a promise chain is an
     arrow function, they probably all should be.
 - Apart from that, newer ES features should be used whenever the author deems them to be appropriate.
-- Flow annotations are welcome and encouraged.
+- While flow annotations are welcome and encouraged consider to migrate the JS file to TypeScript (see next section).
```
Comment (jryans) [inline code_style.md:159]: Let's be a bit stronger here. We're no longer using Flow or adding new annotations really... Perhaps we can just remove this line?

**Review by jryans — CHANGES_REQUESTED** [review]
Thanks for these tweaks! A few comments.

**Review by jryans — APPROVED** [review]
Great, thanks for working on this! 😄

**Pull request merged**